plugins/techdocs-node/src/stages/publish/awsS3.ts

```diff
@@ -475,15 +474,19 @@ export class AwsS3Publish implements PublisherBase {
 
       const entityRootDir = path.posix.join(this.bucketRootPath, entityDir);
 
-      await this.storageClient.send(
-        new HeadObjectCommand({
+      const objectList = await this.storageClient.send(
+        new ListObjectsV2Command({
           Bucket: this.bucketName,
-          Key: `${entityRootDir}/index.html`,
+          Prefix: entityRootDir,
         }),
       );
-      return Promise.resolve(true);
+      return (
+        objectList.Contents?.some(item =>
+          item.Key?.match(/${entityRootDir}\/.*.html$/),
+        ) || false
+      );
     } catch (e) {
-      return Promise.resolve(false);
+      return false;
```

Comment:
I dont think this needed to return a promise because the function is already specified as async.